v03_pipeline/lib/tasks/dataproc/create_dataproc_cluster.py

```diff
@@ -0,0 +1,168 @@
+                    'spark:spark.driver.memory': '41g',
+                    'yarn:yarn.nodemanager.resource.memory-mb': '50585',
+                    'yarn:yarn.scheduler.maximum-allocation-mb': '25292',
+                    'spark:spark.executor.cores': '4',
+                    'spark:spark.executor.memory': '10117m',
+                    'spark:spark.executor.memoryOverhead': '15175m',
+                    'spark:spark.memory.storageFraction': '0.2',
+                    'spark:spark.executorEnv.HAIL_WORKER_OFF_HEAP_MEMORY_PER_CORE_MB': '6323',
+                    'spark:spark.speculation': 'true',
+                    'spark-env:ACCESS_PRIVATE_REFERENCE_DATASETS': '1'
+                    if Env.ACCESS_PRIVATE_REFERENCE_DATASETS
+                    else '0',
+                    'spark-env:CHECK_SEX_AND_RELATEDNESS': '1'
+                    if Env.CHECK_SEX_AND_RELATEDNESS
+                    else '0',
+                    'spark-env:EXPECT_WES_FILTERS': '1'
+                    if Env.EXPECT_WES_FILTERS
+                    else '0',
+                    'spark-env:HAIL_SEARCH_DATA_DIR': Env.HAIL_SEARCH_DATA_DIR,
+                    'spark-env:HAIL_TMP_DIR': Env.HAIL_TMP_DIR,
+                    'spark-env:INCLUDE_PIPELINE_VERSION_IN_PREFIX': '1'
+                    if Env.INCLUDE_PIPELINE_VERSION_IN_PREFIX
+                    else '0',
+                    'spark-env:LOADING_DATASETS_DIR': Env.LOADING_DATASETS_DIR,
+                    'spark-env:PRIVATE_REFERENCE_DATASETS_DIR': Env.PRIVATE_REFERENCE_DATASETS_DIR,
+                    'spark-env:REFERENCE_DATASETS_DIR': Env.REFERENCE_DATASETS_DIR,
+                    'spark-env:CLINGEN_ALLELE_REGISTRY_LOGIN': Env.CLINGEN_ALLELE_REGISTRY_LOGIN,
+                    'spark-env:CLINGEN_ALLELE_REGISTRY_PASSWORD': Env.CLINGEN_ALLELE_REGISTRY_PASSWORD,
+                },
+            },
+            'lifecycle_config': {'idle_delete_ttl': {'seconds': 1200}},
+            'encryption_config': {},
+            'autoscaling_config': {},
+            'endpoint_config': {},
+            'initialization_actions': [
+                {
+                    'executable_file': f'gs://seqr-pipeline-runner-builds/{Env.DEPLOYMENT_TYPE}/{Env.PIPELINE_RUNNER_APP_VERSION}/bin/dataproc_vep_init.bash',
+                    'execution_timeout': {'seconds': 1200},
+                },
+            ],
+        },
+    }
+
+
+@luigi.util.inherits(BaseLoadingPipelineParams)
+class CreateDataprocClusterTask(luigi.Task):
+    # NB: The luigi.dataproc.contrib module was old and bad
+    # so we built our own shim.
+    run_id = luigi.Parameter()
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # https://cloud.google.com/dataproc/docs/tutorials/python-library-example
+        self.client = dataproc.ClusterControllerClient(
+            client_options={
+                'api_endpoint': f'{Env.GCLOUD_REGION}-dataproc.googleapis.com:443'.format(
```

Comment:
do you need the f string and .format here?

Reply:
No I do not!  Good find!

Weird that the linter didn't catch it!

Reply:
it's crazy syntax how it is actually 😵